video_worker/__init__.py

```diff
@@ -96,6 +100,17 @@ def run(self):
             )
 
         self.VideoObject.activate()
+
+        # retry 5 times if the VideoObject is not valid
+        # the encoding celery work queue is thrown before the transaction is finished, so database may not updated yet..
```

Comment:
I don't really understand the reason behind this change. Can you elaborate the scenario you are handling here?

Reply:
When edx-video-worker sends the celery work for encoding, It's meanwhile processing the _ingest_from_s3_bucket, and the Video object was created there, but it's not saved to mysql yet since it's in the middle of transaction. And sometimes, receiving and handling the encoding is happening earlier than the saving, so it needs some synchronization mechanism, or just wait for the Video object is written to mysql on video-pipeline side.

So, I changed to wait for a seconds, but you can change to flush the transaction on edx-video-pipeline side, before send the encoding work queue. But, sleep is much easier and simple, so I've choose this way.

Thanks,

Reply:
@armbox According to your rational, I would say the proper way to fix this issue you are noticing is to update video-pipeline to send the celery task at the right time. Like you mentioned, the right time is after we flush the transaction on database. Keep in mind, the worker do not connect to the mysql database, it does this by doing an API call to the video pipeline webserver for the status of the video. See https://github.com/edx/edx-video-worker/blob/master/video_worker/abstractions.py#L81. This code here would generate up to 5 (which is arbitary and not configurable according to this change) request back to the webserver. 
I say please check how involved it would be to improve the video pipeline code to make sure transaction is complete before the celery task is fired off.

Reply:
Yes, cleaner way to fix this issue is fire celery work after flush the transaction. 

I don't care this fix is adapted or not, but the transaction problem is there. Actually recommend to not adopt it and flush the transaction. To be honest, I do not have knowledge about the django framework, so it's not easy task for me for now.  Please fix the problem anyway you prefer.

Thanks,